### PR TITLE
Change KIWI application builds to build against released containers

### PR DESCRIPTION
**Pull request by dirkmueller**
This is needed so that we have the right oci annotations. Also fixup tests after last change.

### CODE REVIEW

File: src/bci_build/package/appcontainers.py [hunk 1]
```diff
@@ -434,7 +435,11 @@ def _get_nginx_kwargs(os_version: OsVersion):
     ApplicationStackContainer(
         name="helm",
         pretty_name="Kubernetes Package Manager",
-        from_image=f"{_build_tag_prefix(os_version)}/bci-micro:{OsContainer.version_to_container_os_version(os_version)}",
+        from_image=(
```
Comment (alexandrevicenzi) [inline src/bci_build/package/appcontainers.py:438]: nit: can't we simplify this? The only bit that changes is the prefix.

File: src/bci_build/package/__init__.py [hunk 2]
```diff
@@ -868,6 +874,10 @@ def dockerfile_from_line(self) -> str:
     def kiwi_derived_from_entry(self) -> str:
         if self._from_image is None:
             return ""
+        # Adjust for the absolutely braindamaged format that OBS expects to reference
```
Comment (alexandrevicenzi) [inline src/bci_build/package/__init__.py:877]: nit: let's not be that harsh.

```suggestion
        # Adjust for the weird format that OBS expects to reference
```

File: src/bci_build/package/__init__.py [hunk 3]
```diff
@@ -868,6 +874,10 @@ def dockerfile_from_line(self) -> str:
...
+        # external images
+        if self.is_base_container_annotation_available:
+            return f" derived_from=\"obs://SUSE:Registry/standard/{self._from_image.replace('registry.suse.com/','').replace(':', '#')}\""
```
Comment (alexandrevicenzi) [inline src/bci_build/package/__init__.py:880]: nit:  a bit more clear
```suggestion
            # cleanup image name
            image = self._from_image.replace('registry.suse.com/','').replace(':', '#')
            return f" derived_from=\"obs://SUSE:Registry/standard/{image}\""
```

**Review by alexandrevicenzi — APPROVED** [review]

File: src/bci_build/package/__init__.py (hunk 3 above)
Comment (dirkmueller) [reply src/bci_build/package/__init__.py]: good suggestion, merged.

File: src/bci_build/package/appcontainers.py (hunk 1 above)
Comment (dirkmueller) [reply src/bci_build/package/appcontainers.py]: moved into a helper function, since we need it many times.

**Review by alexandrevicenzi — APPROVED** [review]

File: src/bci_build/package/helpers.py [hunk 4]
```diff
@@ -25,3 +28,10 @@ def generate_package_version_check(
 
     return f"""# sanity check that the version from the tag is equal to the version of {pkg_name} that we expect
 {DOCKERFILE_RUN} [ "$(rpm -q --qf '%{{version}}' {pkg_name} | cut -d '.' -f -{cut_count})" = "{pkg_version}" ]"""
+
+
+def generate_from_image(os_version, container_name) -> str:
```
Comment (dcermak) [inline src/bci_build/package/helpers.py:33]: I would also suggest to rename this function to something along the lines `from_image_tag` or `generate_from_image_url` or 

```suggestion
def generate_from_image(os_version: OsVersion, container_name: str) -> str:
    """Return the build tag or url of the container with the name `container_name` for the specified `os_version`."""
```

**Pull request merged**